### ISSUE TITLE
Make flow.kill delete requests without a response

### ISSUE THREAD

**Issue by Gira-X** [mitmproxy/mitmproxy#4711]
I have the use-case where some API calls are taking quite a while sometimes and I would like to drop/kill the request while in transit.

I thought that `flow.kill` (on `X`) would do that, but it always just reports `alert: Killed 0 flows.` when over a request. It seems to only work on intercepted flows but there one also has to wait for the response to arrive.

Proxyman has this feature where the backspace key deletes the request at point even when no response is there yet which is exactly what I would need here.

Am I missing something or is this not implemented yet? `d` also just deletes the flow from view but it continues to be executed in the background.

---

```bash
$ mitmproxy --version
Mitmproxy: 7.0.0
Python:    3.9.6
OpenSSL:   OpenSSL 1.1.1k  25 Mar 2021
Platform:  macOS-10.14.6-x86_64-i386-64bit
```

**Comment by mhils** [comment]
Yes, that's a good point. Killing currently only works from the event hook, we should probably rig it up to work more generally.

**Comment by Prinzhorn** [comment]
Just parking this here

https://github.com/mitmproxy/mitmproxy/blob/fbcffee87f6ca0e8a4bb48123ddd7778ae64a946/mitmproxy/flow.py#L228-L234
